sparkmagic/sparkmagic/controllerwidget/magicscontrollerwidget.py

```diff
@@ -5,21 +5,39 @@
 from sparkmagic.controllerwidget.manageendpointwidget import ManageEndpointWidget
 from sparkmagic.controllerwidget.managesessionwidget import ManageSessionWidget
 from sparkmagic.controllerwidget.createsessionwidget import CreateSessionWidget
+from sparkmagic.livyclientlib.endpoint import Endpoint
+import sparkmagic.utils.configuration as conf
 
 
 class MagicsControllerWidget(AbstractMenuWidget):
     def __init__(self, spark_controller, ipywidget_factory, ipython_display, endpoints=None):
         super(MagicsControllerWidget, self).__init__(spark_controller, ipywidget_factory, ipython_display)
 
         if endpoints is None:
-            endpoints = {}
+            endpoints = {endpoint.url: endpoint for endpoint in self._get_default_endpoints()}
         self.endpoints = endpoints
 
         self._refresh()
 
     def run(self):
         pass
 
+    @staticmethod
+    def _get_default_endpoints():
+        default_endpoints = set()
+        kernel_types = ['python', 'python3', 'scala', 'r']
+
+        for kernel_type in kernel_types:
+            endpoint_config = getattr(conf, 'kernel_%s_credentials' % kernel_type)()
+            if all([p in endpoint_config for p in ["url", "password", "username"]]) and endpoint_config["url"] != "":
+                default_endpoints.add(Endpoint(
```

Comment:
Enpoint now has an auth parameter after the kerberos PR was merged, and the widget should have an extra dropdown now.